core/templates/pages/story-editor-page/modal-templates/new-chapter-title-modal.controller.spec.ts

```diff
@@ -330,18 +336,80 @@ describe('Create New Chapter Modal Controller', function() {
     expect($uibModalInstance.close).not.toHaveBeenCalled();
   }));
 
+  it('should prevent exploration from being added if it contains restricted ' +
+  'interaction types', fakeAsync(function() {
+    $scope.title = 'dummy_title';
+    const invalidStates = ['some_invalid_state'];
+
+    spyOn(
+      editableStoryBackendApiService, 'validateExplorationsAsync'
+    ).and.resolveTo([]);
+    spyOn(curatedExplorationValidationService, 'isExpPublishedAsync')
+      .and.resolveTo(true);
+    spyOn(curatedExplorationValidationService, 'isCorrectnessFeedbackEnabled')
+      .and.resolveTo(true);
+    spyOn(curatedExplorationValidationService, 'isDefaultCategoryAsync')
+      .and.resolveTo(true);
+    spyOn(
+      curatedExplorationValidationService,
+      'getStatesWithRestrictedInteractions').and.resolveTo(invalidStates);
+
+    $scope.saveAsync();
+    flushMicrotasks();
+    $rootScope.$apply();
+
+    expect($scope.statesWithRestrictedInteractions).toBe(invalidStates);
+    expect($uibModalInstance.close).not.toHaveBeenCalled();
+  }));
+
+  it('should prevent exploration from being added if it contains an invalid ' +
+  'multiple choice input', fakeAsync(function() {
+    $scope.title = 'dummy_title';
+    const invalidStates = ['some_invalid_state'];
+
+    spyOn(
+      editableStoryBackendApiService, 'validateExplorationsAsync'
+    ).and.resolveTo([]);
+    spyOn(curatedExplorationValidationService, 'isExpPublishedAsync')
+      .and.resolveTo(true);
+    spyOn(curatedExplorationValidationService, 'isCorrectnessFeedbackEnabled')
+      .and.resolveTo(true);
+    spyOn(curatedExplorationValidationService, 'isDefaultCategoryAsync')
+      .and.resolveTo(true);
+    spyOn(
+      curatedExplorationValidationService,
+      'getStatesWithRestrictedInteractions').and.resolveTo([]);
+    spyOn(
+      curatedExplorationValidationService,
+      'getStatesWithInvalidMultipleChoices').and.resolveTo(invalidStates);
+
+    $scope.saveAsync();
```

Comment:
Please add comment explaining why `saveAsync` is used here.

Reply:
The `saveAsync` function lies in the corresponding `ts` file and is what calls all of the validations and populates all of the error messages for the frontend. All of the tests in this `spec.ts` file basically test the `saveAsync` function. Was something unclear here?

Maybe I can clarify that `saveAsync` is not `applyAsync`.

Reply:
Yep that works